server/src/routes/api.ts

```diff
@@ -4,13 +4,64 @@ import * as mongoose from 'mongoose';
 
 const router: Router = Router();
 
-router.get('/word/list', (req: Request, res: Response) => {
-  WordModel.find()
-    .then((words: Word[]) => res.json(words))
-    .catch(err => res.send(err));
-});
+router.route('/word/list')
+  .get(async (req: Request, res: Response) => {
+    try {
+      const foundList: Word[] = await WordModel.find();
+
+      res.json(foundList);
+    } catch(err) {
+      res.send(err);
+    }
+  })
+
+  .put(async (req: Request, res: Response) => {
+    const updatedList = req.body.list;
+
+    if (!Array.isArray(updatedList)) {
+      return res.status(400).json({'error': 'Bad Data'});
+    }
 
-router.post('/word/add', (req: Request, res: Response) => {
+    for (let i = 0; i < updatedList.length; i++) {
+      WordModel.update({'id': updatedList[i].id}, updatedList[i])
+        .catch(err => res.send(err));
+    }
+
+    WordModel.find()
+      .then((words: Word[]) => res.json(words))
+      .catch(err => res.send(err));
+  })
+
+  .delete(async (req: Request, res: Response) => {
+    const deleteList = req.body.list;
+
+    if (!Array.isArray(deleteList)) {
+      return res.status(400).json({'error': 'Bad Data'});
+    }
+
+    function removeList() {
+      return new Promise((resolve) => {
+        deleteList.forEach((id, index) => {
+          WordModel.findOne({id: id}, null, {limit: 1})
+            .then((word: Word) => {
```

Comment:
Fajny pomysł podoba mi się. Tylko pytanie teraz czy wykorzystać to co jest, czyli loopować, czy po prostu zrobić PUT-a całej zmodyfikowanej listy? Drugie wyjście będzie bardziej eleganckie na pewno. Tylko wtedy po stronie frontendu trzeba by było wysyłać całą nową listę, a nie słowa do usunięcia. To wcześniej działało w taki sposób nie?